src/AzureLoginHelper.ts

```diff
@@ -506,40 +512,39 @@ export class AzureLoginHelper {
 			}
 		}
 
-		if (changed) {
+		if (filtersChanged) {
 			await this.updateConfiguration(azureConfig, resourceFilter);
 		}
 	}
 
-	async noSubscriptionsFound(): Promise<void> {
+	private async noSubscriptionsFound(): Promise<void> {
 		const open: MessageItem = { title: localize('azure-account.open', "Open") };
-		const response = await window.showInformationMessage(localize('azure-account.noSubscriptionsFound', "No subscriptions were found. Set up your account at https://azure.microsoft.com/en-us/free/."), open);
+		const response: MessageItem | undefined = await window.showInformationMessage(localize('azure-account.noSubscriptionsFound', "No subscriptions were found. Set up your account at https://azure.microsoft.com/en-us/free/."), open);
 		if (response === open) {
 			void openUri('https://azure.microsoft.com/en-us/free/?utm_source=campaign&utm_campaign=vscode-azure-account&mktingSource=vscode-azure-account');
 		}
 	}
 
-	private async loadSubscriptions() {
-		const lists = await Promise.all(this.api.sessions.map(session => {
-			const credentials = session.credentials2;
-			const client = new SubscriptionClient(credentials, { baseUri: session.environment.resourceManagerEndpointUrl });
+	private async loadSubscriptions(): Promise<AzureSubscription[]> {
+		const lists: AzureSubscription[][] = await Promise.all(this.api.sessions.map(session => {
+			const credentials: TokenCredentialsBase = session.credentials2;
+			const client: SubscriptionClient = new SubscriptionClient(credentials, { baseUri: session.environment.resourceManagerEndpointUrl });
 			return listAll(client.subscriptions, client.subscriptions.list())
 				.then(list => list.map(subscription => ({
 					session,
 					subscription,
 				})));
 		}));
-		const subscriptions = (<AzureSubscription[]>[]).concat(...lists);
+		const subscriptions: AzureSubscription[] = (<AzureSubscription[]>[]).concat(...lists);
 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 		subscriptions.sort((a, b) => a.subscription.displayName!.localeCompare(b.subscription.displayName!));
 		return subscriptions;
 	}
 
-	private asSubscriptionItems(subscriptions: AzureSubscription[], resourceFilter: string[]): ISubscriptionItem[] {
+	private getSubscriptionItems(subscriptions: AzureSubscription[], resourceFilter: string[]): ISubscriptionItem[] {
 		return subscriptions.map(subscription => {
-			const picked = resourceFilter.indexOf(`${subscription.session.tenantId}/${subscription.subscription.subscriptionId}`) !== -1 || resourceFilter[0] === 'all';
+			const picked: boolean = resourceFilter.indexOf(`${subscription.session.tenantId}/${subscription.subscription.subscriptionId}`) !== -1 || resourceFilter[0] === 'all';
 			return <ISubscriptionItem>{
-				type: 'item',
```

Comment:
Any idea why `type: 'item'` was on here in the first place? Getting rid of it may affect functionality, and I don't really trust the types in this repo to verify if/where it's being used

Reply:
I couldn't find it used anywhere so I'll go ahead and approve, but wouldn't mind you double-checking as well

Reply:
I looked as well and couldn't find any reference to it. I thought `type` might be an artifact from an old version of the `QuickPickItem` interface